### ISSUE TITLE
[1.2.0 Beta 4] Unable to disable "On Demand RRD Updating"

### ISSUE THREAD

**Issue by lux4rd0** [Cacti/cacti#2235]
Good afternoon. Just updated my 1.1.38 instance to 1.2 Beta 4 and have been quite happy with the updates so far (originally started the effort because I'm using php 7.2 and getting a fair number of errors...) With about 75 devices and 2,000 RRDs I was getting about 7-8 seconds on a 30 second poller. Feeling pretty good until I started adding back in some really (slow to respond) Apple Mac devices - which needed 10 second SNMP timeouts to make work. That pushed by polling cycle past the 30 second interval. In order to get some better threading performance, I enabled "On Demand RRD Updating". With the default settings - I started seeing better polling but the RRDs never updated on demand. I would have empty spaces in my graphs. (all defaults left in.) Because of this, I decided to simply turn off the "On Demand RRD Updating". The GUI would show it turned off. But when I went back, it was enabled again. Doing some searching I saw this similar posts:

[https://github.com/Cacti/cacti/issues/1642]
[https://github.com/Cacti/cacti/issues/2100]

They showed the same issue of not being able to disable it but the root cause turned into GUI differences or having remote pollers - and they were in earlier beta versions - non of which were a match for me. I only have one data collector (the Main Poller) using Spine (1.1.38 worked fine but I also compiled the latest 1.2 beta from the development branch.) I have no remote data collectors.

I did find that when the poller did reach the "Maximum Records" or the "How Often Should Boost Update all RRDs" were reached, it said it updated all of the RRDs but only saw data being filled in with interpolated data - not actual data from the poller.

Last but not least - I also noticed that changing spine processes and threads stopped being reflected accurately no matter what lever I pulled. Adding more processes or threads never got reflected in the logging. (However, if I ran the spine poller via the command line with "./spine -R" it always showed the stats correctly. (May do some more research on that and potentially open a different ticket on that...)

```
2018/12/27 16:11:17 - RECACHE STATS: Poller: RecacheTime:0.9055 DevicesRecached:1
2018/12/27 16:11:16 - SYSTEM BOOST STATS: Time:0.5600 RRDUpdates:20037
2018/12/27 16:11:16 - SYSTEM STATS: Time:14.2874 Method:spine Processes:1 Threads:20 Hosts:87 HostsPerProcess:87 DataSources:1932 RRDsProcessed:0
2018/12/27 16:10:46 - SYSTEM STATS: Time:14.2679 Method:spine Processes:1 Threads:20 Hosts:87 HostsPerProcess:87 DataSources:1932 RRDsProcessed:0
```

Waiting for On Demand to update: (notice the empty data in the RRD)

![screen shot 2018-12-27 at 4 11 08 pm](https://user-images.githubusercontent.com/30187533/50495979-50921f80-09f2-11e9-93e3-19a3b2bceb8c.png)

![cacti_graph_61_rra5](https://user-images.githubusercontent.com/30187533/50495982-57b92d80-09f2-11e9-87f3-58de5b82f4b9.png)

After the RRDUpdates:

![screen shot 2018-12-27 at 4 12 38 pm](https://user-images.githubusercontent.com/30187533/50495995-6f90b180-09f2-11e9-9dc0-5a102d05f0b1.png)

![cacti_graph_61_rra5 1](https://user-images.githubusercontent.com/30187533/50496007-80412780-09f2-11e9-8dad-85a193cfc615.png)

Long story short - I'd like to be able to simply disable this on demand boost functionality but I can't. Since I can't - I'd like to see if the data can actually be updated correctly after the thresholds have been met. For now I can simply set "Maximum Records" to 1 and it correctly updates the RRDs with every 30 second polling - but that totally defeats the purpose of using Boost. :)

Also - if this is something I should toss into the Cacti forums instead of this bugs forum, happy to do so as well.

Running CentOS 7.6, Apache/2.4.6 (CentOS) PHP/7.2.13

Thanks!

_DS

**Comment by lux4rd0** [comment]
Sure - just seeing now that the Data Collector Information has:

**Processes**: 1
**Threads**: 20

_(Which is what the System Stats shows)_

I only have the one poller.

I read the hover over tool tips and see that the settings that I've been changing have moved. :) Interesting that running spine via CLI pulls the settings differently.

**Comment by cigamit** [comment]
When you attempt to disable the boost, it will actually stay enabled until the boost cache is totally drained.  you might want to force a run from the command line using 'php -q poller_boost.php --force'.  That should speed things up.

**Comment by cigamit** [comment]
The solution, in the near term if this is not working is to first not have any remove data collectors, the second would be to truncate the boost table between updates if boost is just taking too long, disable the poller, drain the boost cache, and then re-enable the poller.

**Comment by lux4rd0** [comment]
I appreciate the details as when I did drop the Maximum Records to a low number I was able to go back and disable the boost before my poller ran in the next 30 seconds. I would make an assumption that there would be situations where the cache would never be 100% drained and therefor never able to be disabled without getting into the database. Maybe some UI updates on the status of an empty cache when choosing to disable it. Or when choosing to disable it, it automatically flushes the boost cache and disables. It felt like a bug when I'd disable it - it said it was disabled, and then when I went back it was "still" enabled. :) Thank you for the details on how it's put together. I'd say the my original "unable to disable" issue is solved - there's still the notion of the cached data not being inserted correctly (or the data being incorrect) while I was testing that functionality. But that's probably another issue to be opened.

**Comment by interduo** [comment]
I appreciate Your work about that issue, I can suggest that there should be also opportunity to do RRD Updating at specified time (for example in crontab by cli).

**Comment by TheWitness** [comment]
@interduo, log a separate feature for that.  Also, @lux4rd0 please check the lastest boost in 1.2.11 or the 1.2.x branch.  I think you will find it much better performing.  Again, with multiple data collectors, as long as they are defined in the system, you can not disable boost.  I'm going to close this one though.
